logic/account.js

```diff
@@ -596,19 +650,56 @@ Account.prototype.getAll = function (filter, fields, cb) {
 		limit: limit,
 		offset: offset,
 		sort: sort,
-		alias: 'a',
 		condition: filter,
-		fields: realFields
+		fields: realFields,
+		alias: 'a',
+		join: {
+			delegates: {
+				type: 'left',
+				on: {
+					'a.address': 'd.address'
+				},
+				alias: 'd'
+			}
+		}
 	});
 
 	this.scope.db.query(sql.query, sql.values).then(function (rows) {
+		var lastBlock = modules.blocks.lastBlock.get();
+		// If the last block height is undefined, it means it's a genesis block with height = 1
+		var totalSupply = __private.blockReward.calcSupply(lastBlock.height || 0);
+
+		if (defaultFields || fields.includes('approval')) {
+			rows.map(function (accountRow) {
+				accountRow.approval = __private.calculateApproval(accountRow.vote, totalSupply);
+			});
+		}
+
+		if (defaultFields || fields.includes('productivity')) {
+			rows.map(function (accountRow) {
+				accountRow.productivity = __private.calculateProductivity(accountRow.producedBlocks, accountRow.missedblocks, accountRow.rank);
+			});
+		}
+
 		return setImmediate(cb, null, rows);
 	}).catch(function (err) {
 		library.logger.error(err.stack);
 		return setImmediate(cb, 'Account#getAll error');
 	});
 };
 
+
+__private.calculateApproval = function (votersBalance, totalSupply) {
+	var approval = (votersBalance / totalSupply) * 100;
+	return Math.round(approval * 1e2) / 1e2;
+};
+
+__private.calculateProductivity = function (producedBlocks, missedBlocks, rank) {
+	var outsider = rank + 1 > slots.delegates;
+	var percent = 100 - (missedBlocks / ((producedBlocks + missedBlocks) / 100));
```

Comment:
if `producedBlocks` and `missedBlocks` are both 0 we are dividing by 0 and receiving `NaN` as the result